docs/filemanagement.rst

```diff
@@ -208,6 +229,54 @@ like so:
 
     content_string = file4.GetContentString(encoding='ISO-8859-1')
 
+Upload data as bytes in memory buffer
+--------------------------------------
+
+Data can be kept as bytes in an in-memory buffer when we use the io module’s
+Byte IO operations, we can upload files that reside in memory, for
+example we have a base64 image, we can decode the string and upload it to drive
+without the need to save as a file and use `SetContentFile(filename)`_
+
+.. code-block:: python
+
+    import io
+    from pydrive2.drive import GoogleDrive
+
+    # create GoogleDrive instance with authenticated GoogleAuth instance.
+    drive = GoogleDrive(gauth)
+    # define file name and type
+    metadata = {
+        'title': 'image_test',
+        'mimeType': 'image/jpeg'
+    }
+    # create file
+    file = drive.CreateFile(metadata=metadata)
+    # Buffered I/O implementation using an in-memory bytes buffer.
+    image_file = io.BytesIO(image_bytes)
+    # set the content of the file
+    file.content = image_file
+    # upload the file to google drive
+    file.Upload()
+
+Upload file to specific folder
+--------------------------------------
+
+In order to upload a file into a specific drive folder we need to pass the
+id of the folder in the metadata param from `CreateFile()`_.
+Save the image from the previous example into a specific folder
+
+.. code-block:: python
+
+    metadata = {
+        'parents': [
+            {"kind": "drive#fileLink", "id": id_drive_folder}
+        ],
+        'title': 'image_test',
+        'mimeType': 'image/jpeg'
+    }
+    # create file
+    file = drive.CreateFile(metadata=metadata)
```

Comment:
I think you need to run `Upload` at the end?

Reply:
it´s a simple example to show how to add the param to metadata dict, but if you want i will put the full example to be explicit

Reply:
the example name is "Upload file to specific folder", but it doesn't actually upload it (it just creates a wrapper). I might mislead users (do we need to anything else, or not?)

It won't be more complicated, I hope one line should be enough. And considering that we'll probably cleanup the `params` structure it will the same length or even less overall.